### PR TITLE
🐛 fixup Makefile, should fix image build on master

### PR DESCRIPTION
**Pull request by sbueringer**
**What this PR does / why we need it**:

Syncs the Makefile with CAPA and should fix our build on master

Also: 
As we're not really (imho) supporting migrations from v1alpha2 to v1alpha3 and the v1alpha2 package was unused in our implementation I would suggest removing it. I'm happy when we have v1alpha3 finalized I don't think we have the bandwidth to support migrations explicitly right now

See also: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/499

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -380,7 +370,7 @@ create-cluster: $(CLUSTERCTL) $(ENVSUBST) ## Create a development Kubernetes clu
 
 	# Deploy calico
 	kubectl --kubeconfig=./kubeconfig apply -f https://docs.projectcalico.org/manifests/calico.yaml
-	# FIXME:
+
 	# Create a worker node with MachineDeployment.
 	kubectl apply -f examples/_out/machinedeployment.yaml
```
Comment (hidekazuna) [inline Makefile]: Please update v0.3.0-rc.3 to v0.3.0 in line 340.

File: docs/config.md [hunk 2]
```diff
@@ -201,7 +201,7 @@ spec:
 1. For example in `examples/_out/controlplane.yaml` set `spec.rootVolume.diskSize` to something greater than `0` means boot from volume.
 
    ```yaml
-   apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+   apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
    kind: OpenStackMachine
    metadata:
      name: <cluster-name>-controlplane-0
```
Comment (hidekazuna) [inline docs/config.md]: same here.

File: docs/config.md [hunk 3]
```diff
@@ -183,7 +183,7 @@ spec:
 Instead of tagging, you also have the option to add metadata to instances. This functionality should be more commonly available than tagging. Here is a usage example:
 
 ```yaml
-apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
 kind: OpenStackMachine
 metadata:
   name: <cluster-name>-controlplane-0
```
Comment (hidekazuna) [inline docs/config.md]: same here.

File: docs/config.md [hunk 4]
```diff
@@ -96,12 +96,12 @@ You can reference which operating system image you want to use in the `controlpl
...
+If you have a complex query that you want to use to lookup a network, then you can do this by using a network filter. More details about the filter can be found in [NetworkParam](../api/v1alpha3/types.go)
 
 By using filters to look up a network, please note that it is possible to get multiple networks as a result. This should not be a problem, however please test your filters with `openstack network list` to be certain that it returns the networks you want. Please refer to the following usage example:
 
 ```yaml
-apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
 kind: OpenStackMachine
 metadata:
   name: <cluster-name>-controlplane-0
```
Comment (hidekazuna) [inline docs/config.md]: Please remove -0.

File: docs/config.md [hunk 5]
```diff
@@ -117,7 +117,7 @@ spec:
 You can specify multiple networks (or subnets) to connect your server to. To do this, simply add another entry in the networks array. The following example connects the server to 3 different networks:
 
 ```yaml
-apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
 kind: OpenStackMachine
 metadata:
   name: <cluster-name>-controlplane-0
```
Comment (hidekazuna) [inline docs/config.md]: same here.

File: docs/config.md [hunk 6]
```diff
@@ -136,7 +136,7 @@ spec:
 Rather than just using a network, you have the option of specifying a specific subnet to connect your server to. The following is an example of how to specify a specific subnet of a network to use for your server.
 
 ```yaml
-apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
 kind: OpenStackMachine
 metadata:
   name: <cluster-name>-controlplane-0
```
Comment (hidekazuna) [inline docs/config.md]: same here.

File: docs/config.md [hunk 7]
```diff
@@ -168,7 +168,7 @@ spec:
 To tag resources specific to a machine, add a value to the tags field in `controlplane.yaml` and `machinedeployment.yaml` like this:
 
 ```yaml
-apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha3
 kind: OpenStackMachine
 metadata:
   name: <cluster-name>-controlplane-0
```
Comment (hidekazuna) [inline docs/config.md]: same here.

File: Makefile (hunk 1 above)
Comment (sbueringer) [reply Makefile]: Done

File: docs/config.md (hunk 2 above)
Comment (sbueringer) [reply docs/config.md]: Removed all -0 in this file

**Comment by sbueringer** [comment]
@hidekazuna all done. Thx for the review

**Comment by hidekazuna** [comment]
@sbueringer Thanks, confirmed.

**Comment by prankul88** [comment]
@sbueringer Will review this tomorrow asap.

**Comment by jichenjc** [comment]
looks good, just need other folks to take a quick look

**Comment by prankul88** [comment]
@sbueringer Can you update the version of version of kubebuilder https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/b406c51d99841d707b17f9965ca7f5ae05448a2a/config/crd/bases/infrastructure.cluster.x-k8s.io_openstackclusters.yaml#L7 to v0.2.6 in all the yaml. Thanks

**Comment by prankul88** [comment]
Otherwise LGTM

File: Makefile [hunk 8]
```diff
@@ -295,20 +298,7 @@ release: clean-release  ## Builds and push container images using the latest git
 
 .PHONY: release-manifests
 release-manifests: $(RELEASE_DIR) ## Builds the manifests to publish with a release
-	kustomize build config > $(RELEASE_DIR)/infrastructure-components.yaml
-
-.PHONY: release-binary
```
Comment (prankul88) [inline Makefile]: @sbueringer Was curious to know why do we not need `release-binary` in `release`

**Comment by sbueringer** [comment]
> @sbueringer Can you update the version of kubebuilder
> 
> https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/b406c51d99841d707b17f9965ca7f5ae05448a2a/config/crd/bases/infrastructure.cluster.x-k8s.io_openstackclusters.yaml#L7
> 
> to v0.2.6 in all the yaml. Thanks

I would like to keep it as is. The acutal version is defined in hack/tools/go.mod and is sync with CAPA (controller-tools). Not sure what consequences it have to use something else so I would just like to keep it sync for now

File: Makefile (hunk 8 above)
Comment (sbueringer) [reply Makefile]: Not sure why it was added. release-binary is used in CAPA to build clusterawsadm. But we don't have a binary like this for now, so we don't need the target.

**Comment by sbueringer** [comment]
@hidekazuna @jichenjc @prankul88 
PTAL :)

File: Makefile (hunk 8 above)
Comment (prankul88) [reply Makefile]: Perfect. Thanks

**Comment by prankul88** [comment]
LGTM. Others can confirm. :)

**Comment by jichenjc** [comment]
looks good, let's try use it then submit further PRs :)

**Comment by sbueringer** [comment]
> looks good, let's try use it then submit further PRs :)

Perfect :). Good news is I have time today to finalize the template PR. After that one we should have an almost final v1alpha3 version

**Comment by hidekazuna** [comment]
Cluster was created successfully in my env.

/lgtm

**Pull request merged**